### ISSUE TITLE
Import table sometimes doesn't auto-refresh to show new table

### ISSUE THREAD

**Issue by snickell** [code-dot-org/code-dot-org#58456]
We probably need to look at the JSX logic for when it calls a refresh

![Image](https://github.com/code-dot-org/code-dot-org/assets/223277/280dbed1-42b3-4ad9-9d78-7e918d6aa7f9)

**Comment by snickell** [comment]
This repros consistently on prod for me:

https://github.com/user-attachments/assets/78b9c0d6-fddf-4864-b43c-6f3c8d4cfa56

**Comment by snickell** [comment]
Locally it **usually** does NOT repro:

https://github.com/user-attachments/assets/35c37a01-e1e2-47e1-a064-2bb0bd76a76f

But I was able to get it to repro locally by trying repeatedly many times. Looks very likely to be a timing related bug, probably we're not doing a refresh of the redux store AFTER the async add shared table or whatever succeeds.

**Comment by snickell** [comment]
I was not able to get a consistent local repro consistent by introducing 3s of latency into all requests using Chrome devtools.

What the latency made very clear, however, is that our `get_table_names` call (refresh) is happening in parallel with `add_shared_table`:
![Image](https://github.com/user-attachments/assets/9b35b8b5-db00-42ee-8b17-d0732f48bead)

Contrast with [Delete], which works consistently, and the calls are (properly) sequential:
![Image](https://github.com/user-attachments/assets/cc4f9630-dd40-488c-a41b-37ebce502c85)

**Comment by snickell** [comment]
Adding a 1s timeout before calling the server inside DatablockStorage.addSharedTable makes the bug repro 100% consistently locally:
```
DatablockStorage.addSharedTable = function (tableName, onSuccess, onError) {
  setTimeout(() => {
    _fetch('add_shared_table', 'POST', {
      table_name: tableName,
    }).then(onSuccess, onError);
  }, 1000);
};
```

**Comment by snickell** [comment]
Fixed by #59799
